src/crimson/osd/main.cc

```diff
@@ -208,6 +209,12 @@ int main(int argc, char* argv[])
     usage(argv[0]);
     return EXIT_SUCCESS;
   }
+  if (boost::algorithm::ends_with(argv[0], "/ceph-osd")) {
+    // HACK: we ship crimson-osd as ceph-osd, so we shouldn't require
+    // users to restrict the no of CPUs Seastar will sit on.
+    app_args.emplace_back("--smp");
+    app_args.emplace_back("1");
```

Comment:
i'd suggest extract this into a separate function and guard it with something like
```c++
void override_seastar_opts(std::vector<const char*>& args)
{
#ifndef CRIMSON_M2N
  // TODO:we don't have a way to communicate the resource requirements 
  // with the deployment tools, like cephadm and rook, which don't set, for 
  // instance, aio-max-nr for us. but we should fix this, once crimson is able
  // to run on a multi-core system, i.e., once m-to-n problem is resolved.
  args.emplace_back("--smp");
  args.emplace_back("1");
#endif
}
```